boot/bootutil/include/bootutil/crypto/ecdh_x25519.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * This module provides a thin abstraction over some of the crypto
+ * primitives to make it easier to swap out the used crypto library.
+ *
+ * At this point, there are two choices: MCUBOOT_USE_MBED_TLS, or
+ * MCUBOOT_USE_TINYCRYPT.  It is a compile error there is not exactly
+ * one of these defined.
+ */
+
+#ifndef __BOOTUTIL_CRYPTO_ECDH_X25519_H_
+#define __BOOTUTIL_CRYPTO_ECDH_X25519_H_
+
+#include "mcuboot_config/mcuboot_config.h"
+
+#if (defined(MCUBOOT_USE_TINYCRYPT)) != 1
+    #error "One crypto backend must be defined: TINYCRYPT"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// TODO: also || MBDED_TLS 
```

Comment:
Also this comment should be `/* ... */`, and just to mention it, mbed TLS happens to have imported x25519 from the Everest project some time ago: https://github.com/ARMmbed/mbedtls/blob/development/3rdparty/everest/library/x25519.c. So this can be updated later, in another PR.

Reply:
I realized the X25519 implementation is sourced from `fiat-crypto` so I've also added it for mbedtls and removed that comment.